lib/rex/proto/proxy/socks5.rb

```diff
@@ -0,0 +1,543 @@
+
+# -*- coding: binary -*-
+#
+# sf - Sept 2010
+# surefire - April 2018
+#
+# TODO: Support SOCKS authentication (just accept anything, I guess)
```

Comment:
Is there a risk when this is bound to a loopback port? Do most SOCKS5 clients support auth? I presume basic would be sufficient.

Reply:
I'm personally not concerned about the risk of a listener on loopback.  That said:

1) We listen on global (0.0.0.0) by default.  Let's change that to 127.0.0.1.
2) If we are paranoid for some reason, we could randomly generate creds and print them when the module starts.  For example:

> ```
> msf5 auxiliary(server/socks5) > run
> [*] Stopping the socks5 proxy server
> [*] Auxiliary module running as background job 1.
> [*] Starting the socks5 proxy server
> [*] SOCKS5 proxy at 127.0.0.1:1080 with username 'msf' and password 'VLCew6Mz'
> ```

Personally, I think this is overkill, but I'd like a vote:  :+1: for generate creds, and :-1: for not.

Regardless of the vote, I'll create options to specify a required username and password.